app/file-upload/server/lib/FileUpload.js

```diff
@@ -66,7 +66,7 @@ export const FileUpload = {
 		const room = Rooms.findOneById(file.rid);
 		const directMessageAllowed = settings.get('FileUpload_Enabled_Direct');
 		const fileUploadAllowed = settings.get('FileUpload_Enabled');
-		if (canAccessRoom(room, user, file) !== true) {
+		if (canAccessRoom(room, user, file) !== true && user.type !== 'app') {
```

Comment:
`canAccessRoom` is a heavy operation, if app users will bypass it, we can bypass calling it:
```suggestion
		if (user.type !== 'app' && canAccessRoom(room, user, file) !== true) {
```